planner/core/rule_predicate_push_down.go

```diff
@@ -157,6 +157,7 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 		tempCond = append(tempCond, p.OtherConditions...)
 		tempCond = append(tempCond, predicates...)
 		tempCond = expression.ExtractFiltersFromDNFs(p.ctx, tempCond)
+		p.SetCartesianJoin(tempCond)
```

Comment:
IMO, we should check `Left`/`Right`/`Equal` conditions, `Other` Conditions should not be checked even it's one of the join condition. Because it's still required to perform a cartesian join firstly.